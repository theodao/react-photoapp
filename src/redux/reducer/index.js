import { combineReducers } from 'redux';

export default combineReducers({
  auth: require('./authReducer').reducer,
});
