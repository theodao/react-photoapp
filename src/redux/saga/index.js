import { all } from 'redux-saga/effects';

/** All saga  */
import authSaga from './authSaga';
import categorySaga from './categorySaga';
import appSaga from './appSaga';

export default function* root() {
  yield all([...authSaga, ...categorySaga, ...appSaga]);
}
