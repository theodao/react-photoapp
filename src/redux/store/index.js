import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import logger from 'redux-logger';
import rootReducer from '../reducer';

export default () => {
  /** Redux configuration  */
  const middleware = [];
  const enhancers = [];

  /** Saga middleware  */
  const sagaMiddleware = createSagaMiddleware();
  middleware.push(sagaMiddleware);

  /** Logger middleware  */
  middleware.push(logger);

  enhancers.push(applyMiddleware(...middleware));

  /** Store configuration  */
  const store = createStore(rootReducer, compose(...enhancers));

  /** Run root saga  */
  // const sagaManager = sagaMiddleware.run();

  return {
    store,
    // sagaManager,
  };
};
