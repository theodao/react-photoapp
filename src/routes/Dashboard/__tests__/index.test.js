import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import { Dashboard } from '../';

describe('Testing dashboard route', () => {
  afterEach(cleanup);

  const store = createStore(() => ({
    auth: {
      isLoggedIn: false,
    },
    app: {
      isLoading: false,
    },
  }));

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Dashboard
          category={{
            isFetching: false,
            categories: [],
          }}
          auth={{
            isLoggedIn: false,
          }}
        />
        ,
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // it('Should call api', () => {
  //   const mockFetchingData = jest.fn();

  //   const tree = renderer.create(
  //     <Provider store={store}>
  //       <Dashboard
  //         category={{
  //           isFetching: false,
  //           categories: [],
  //         }}
  //         auth={{
  //           isLoggedIn: false,
  //         }}
  //         fetchCategoryList={mockFetchingData}
  //       />
  //     </Provider>,
  //   );
  //   tree.update();

  //   expect(mockFetchingData).toHaveBeenCalled();
  // });
});
