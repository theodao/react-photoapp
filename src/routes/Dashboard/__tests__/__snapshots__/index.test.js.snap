// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing dashboard route Should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="height: 100%;"
  >
    <div
      style="display: flex; justify-content: space-between;"
    >
      <div
        style="margin-left: 10px;"
      >
        <div
          class="TopMenu u-backgroundWhite u-paddingVerticalExtraSmall"
        >
          <div
            class="TopMenu-list"
          >
            <div
              class="TopMenu-item u-positionRelative is-active u-flexInline u-alignItemsCenter u-cursorPointer"
            >
              <div
                class="TopMenu-itemBefore u-positionAbsolute u-positionLeft u-positionTop u-backgroundTransparent u-widthFull"
              />
              <div
                class="TopMenu-itemAfter u-positionAbsolute u-positionLeft u-positionBottom u-backgroundPrimary u-widthFull"
              />
              <span
                class="u-positionRelative u-flexInline u-flexGrow-1 u-paddingVerticalTiny hover:u-textDecorationNone u-textPrimary"
              >
                <div
                  class="u-flexGrow-1"
                >
                  Dashboard
                </div>
              </span>
            </div>
            <div
              class="TopMenu-item u-positionRelative u-marginLeftLarge u-flexInline u-alignItemsCenter u-cursorPointer"
            >
              <a
                class="u-positionRelative u-flexInline u-flexGrow-1 u-paddingVerticalTiny hover:u-textDecorationNone u-textDark hover:u-textPrimary"
                href="#"
                role="button"
              >
                <div
                  class="u-flexGrow-1"
                >
                  Add categories
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div>
        <button
          class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone u-cursorPointer u-textUppercase"
          style="margin: 10px; font-size: 0.9rem;"
        >
          <span
            class="Button-label u-inlineBlock"
          >
            Login
          </span>
        </button>
        <button
          class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textGray hover:u-textGray u-backgroundLightest hover:u-backgroundUltraLight u-border hover:u-textDecorationNone u-cursorPointer u-textUppercase"
          style="margin: 10px; font-size: 0.9rem;"
        >
          <span
            class="Button-label u-inlineBlock"
          >
            Sign up
          </span>
        </button>
      </div>
    </div>
    <div>
      <div
        class="sc-AxiKw gARtGm"
      >
        <div
          class="sc-AxjAm dNrbJL"
        />
        <div
          style="max-width: 1140px; display: flex; justify-content: center; margin: 0px auto;"
        >
          <div
            class="EmptyState u-inlineBlock u-textCenter"
          >
            <img
              alt="General"
              class="u-maxWidthFull"
              src="https://designsystem-cdn.got-it.ai/images/general.8da83c28.svg"
              width="240"
            />
            <div
              class="EmptyState-heading u-marginTopSmall u-text650 u-textLight"
            >
              Category Empty
            </div>
          </div>
        </div>
        <div
          class="sc-AxjAm dNrbJL"
        />
        <div
          class="u-textCenter"
        >
          <ul
            class="Pagination u-marginNone u-paddingNone"
          >
            <li
              class="Pagination-item u-inlineBlock u-marginHorizontalTiny Pagination-prev"
            >
              <span
                class="u-block hover:u-textDecorationNone"
              >
                <div
                  class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textGray hover:u-textGray u-backgroundLightest hover:u-backgroundUltraLight u-border hover:u-textDecorationNone Button--small is-disabled u-cursorNotAllow u-pointerEventsNone u-fontMedium"
                >
                  <span
                    class="Button-icon u-inlineBlock"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      focusable="false"
                      height="16px"
                      style="display: inline-block; vertical-align: middle;"
                      viewBox="0 0 24 24"
                      width="16px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m10.0145413 11.9953152 6.6183511-6.80357144c.4894768-.50357143.4894768-1.31785715 0-1.81607143-.4894768-.50357143-1.2809712-.49821429-1.770448 0l-7.49836789 7.70892857c-.4738552.4875-.4842696 1.2696429-.0364504 1.7732143l7.52961109 7.7625c.2447384.2517857.5675848.375.885224.375s.6404856-.1232143.885224-.375c.4894768-.5035714.4894768-1.3178572 0-1.8160714z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </span>
                </div>
              </span>
            </li>
            <li
              class="Pagination-item is-active u-inlineBlock u-marginHorizontalTiny"
            >
              <span
                class="u-block hover:u-textDecorationNone"
              >
                <div
                  class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone Button--small u-cursorPointer u-fontMedium"
                >
                  1
                </div>
              </span>
            </li>
            <li
              class="Pagination-item u-inlineBlock u-marginHorizontalTiny Pagination-next"
            >
              <span
                class="u-block hover:u-textDecorationNone"
              >
                <div
                  class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textGray hover:u-textGray u-backgroundLightest hover:u-backgroundUltraLight u-border hover:u-textDecorationNone Button--small is-disabled u-cursorNotAllow u-pointerEventsNone u-fontMedium"
                >
                  <span
                    class="Button-icon u-inlineBlock"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      focusable="false"
                      height="16px"
                      style="display: inline-block; vertical-align: middle;"
                      viewBox="0 0 24 24"
                      width="16px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m13.9854587 11.9966505-6.6183511-6.80610342c-.4894768-.50375884-.4894768-1.3183476 0-1.8167473.48947679-.49839971 1.28097118-.49839971 1.77044798 0l7.49836792 7.71179752c.4738552.4876814.4842696 1.2701154.0364504 1.7738742l-7.52961112 7.7653889c-.2447384.2518795-.5675848.3751396-.88522399.3751396-.3176392 0-.6404856-.1232601-.88522399-.3751396-.4894768-.5037588-.4894768-1.3183476 0-1.8167473z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </span>
                </div>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  ,
</DocumentFragment>
`;

exports[`Testing dashboard route testing add item modal snapshot 1`] = `<DocumentFragment />`;

exports[`Testing dashboard route testing image modal snapshot 1`] = `<DocumentFragment />`;
